Source/Core/Rectangle.js

```diff
@@ -872,6 +874,206 @@ define([
         return result;
     };
 
+    var unprojectedScratch = new Cartographic();
+    var cornerScratch = new Cartographic();
+    var projectedScratch = new Cartesian3();
+    /**
+     * Approximates a Cartographic rectangle's extents in some map projection by projecting
+     * points in a grid throughout the rectangle.
+     *
+     * @function
+     *
+     * @param {Rectangle} cartographicRectangle An input rectangle in geographic coordinates.
+     * @param {MapProjection} mapProjection A MapProjection indicating a projection from geographic coordinates.
+     * @param {Rectangle} [result] Rectangle on which to store the projected extents of the input.
+     * @param {Number} [steps=8] Number of points to sample along each side of the geographic Rectangle.
+     */
+    Rectangle.approximateProjectedExtents = function(cartographicRectangle, mapProjection, result, steps) {
+        //>>includeStart('debug', pragmas.debug);
+        Check.defined('cartographicRectangle', cartographicRectangle);
+        Check.defined('mapProjection', mapProjection);
+        //>>includeEnd('debug');
+
+        result = defaultValue(result, new Rectangle());
+        steps = defaultValue(steps, 8);
+
+        result.west = Number.MAX_VALUE;
+        result.east = -Number.MAX_VALUE;
+        result.south = Number.MAX_VALUE;
+        result.north = -Number.MAX_VALUE;
+
+        var geographicCorner = Rectangle.southwest(cartographicRectangle, cornerScratch);
+        var geographicWidth = cartographicRectangle.width;
+        var geographicHeight = cartographicRectangle.height;
+
+        var geographicWidthStep = geographicWidth / (steps - 1);
+        var geographicHeightStep = geographicHeight / (steps - 1);
+
+        var projected = projectedScratch;
+        var unprojected = unprojectedScratch;
+
+        for (var longIndex = 0; longIndex < steps; longIndex++) {
+            for (var latIndex = 0; latIndex < steps; latIndex++) {
+                unprojected.longitude = geographicCorner.longitude + geographicWidthStep * longIndex;
+                unprojected.latitude = geographicCorner.latitude + geographicHeightStep * latIndex;
+
+                mapProjection.project(unprojected, projected);
```

Comment:
I feel like the project and setting the rectangle are being done enough times in this file to use a helper function.

Reply:
I'll update for `unproject` below.